channels/audin/server/audin.c

```diff
@@ -132,29 +153,39 @@ static void audin_server_send_formats(audin_server* audin, wStream* s)
 		if (audin->context.server_formats[i].cbSize)
 		{
 			if (!Stream_EnsureRemainingCapacity(s, audin->context.server_formats[i].cbSize))
-				return;
+			{
+				WLog_ERR(TAG, "Stream_EnsureRemainingCapacity failed!");
+				return CHANNEL_RC_NO_MEMORY;
+			}
 
 			Stream_Write(s, audin->context.server_formats[i].data,
 				audin->context.server_formats[i].cbSize);
 		}
 	}
 
-	WTSVirtualChannelWrite(audin->audin_channel, (PCHAR) Stream_Buffer(s), Stream_GetPosition(s), &written);
+	return WTSVirtualChannelWrite(audin->audin_channel, (PCHAR) Stream_Buffer(s), Stream_GetPosition(s), &written) ? CHANNEL_RC_OK : ERROR_INTERNAL_ERROR;
```

Comment:
This implicit cast from BOOL to WIN32ERROR is valid but the error code (0x01 for false) would mean ERROR_INVALID_FUNCTION.